go.mod

```diff
@@ -3,7 +3,9 @@ module k8s.io/kube-state-metrics
 require (
 	github.com/brancz/gojsontoyaml v0.0.0-20190425155809-e8bd32d46b3d
 	github.com/campoy/embedmd v1.0.0
+	github.com/coreos/prometheus-operator v0.34.0
```

Comment:
This is a rather heavy dependency. I'd actually prefer either of the two approaches below:

i) Move the MultiListWatcher to its own project/module.
OR
ii) We could copy the implementation code over to this project. If we choose this approach, we can tweak it a bit to use `klog` instead of a NoOp logger implementation of a foreign (in this project's context) log project.

Reply:
Preference for approach 2.

Reply:
i) I don't think we would move this into its own project from the prometheus-operator as we in prometheus-operator would find it hard to maintain that and bump deps there etc.

ii) so someone, I think istio did this :D with this exact pkg, they copied it over to their project a while ago, but never kept up with our bug fixes and they ended up with a huge bug we fixed. What I mean is, we would need to maintain both pkgs then.

While I agree being dependent on prom-operator is a huge thing, I would like to see how much code we would have to maintain in that pkg in here? @brancz how much work do you think this would be to move it to ksm?

Reply:
It’s already a well encapsulated package there so copying it over is trivial.